test/refactoring/ConvertAsync/convert_result.swift

```diff
@@ -0,0 +1,175 @@
+func simple(_ completion: (Result<String, Error>) -> Void) { }
+func noError(_ completion: (Result<String, Never>) -> Void) { }
+
+// RUN: %refactor -convert-call-to-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix=SIMPLE %s
+simple { res in
+  print("result \(res)")
+}
+// SIMPLE: let res = try await simple()
+// SIMPLE-NEXT: print("result \(<#res#>)")
+
+// RUN: %refactor -convert-call-to-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix=NOERROR %s
+noError { res in
+  print("result \(res)")
+}
+// NOERROR: let res = await noError()
+// NOERROR-NEXT: print("result \(<#res#>)")
+
+// RUN: %refactor -convert-call-to-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix=DOBLOCK %s
+simple { res in
+  print("before")
+  switch res {
+  case .success(let str):
+    print("result \(str)")
+  case .failure(let err):
+    print("error \(err)")
+  }
```

Comment:
It would be good to also add an example with an explicit 'break' in the cases to see what happens to it. An example with multiple success or failure cases, e.g. with both `case .success(let str) where str.starts(with: "a"):` and `case .success(let str):` would be good to check too, as well as one with a default. I'm not saying we need to support all those, but just to check we do the fallback/fail if we should be doing that.